deployment/actionManger.go

```diff
@@ -298,3 +407,11 @@ func (mgr *ActionManager) deactivateIfHashDiffers(w *worker.V9Worker, compID wor
 
 	return nil
 }
+
+func (mgr *ActionManager) updateInstanceCount(compPath worker.ComponentPath, instanceCount int) {
```

Comment:
This function is a huge race condition. Please extend the existing channel system here